src/ast/codegen_llvm.cpp

```diff
@@ -854,48 +859,70 @@ void CodegenLLVM::visit(Binop &binop)
 
       expr_ = b_.CreateStrcmp(left_string, right_string, inverse);
     }
+    expr_ = b_.CreateIntCast(expr_, b_.getInt64Ty(), false);
```

Comment:
We can declare `do_signed` earlier (default to false) and keep the uppercast in only one place (at the end of the function). This will also avoid the space changes below.